meilisearch-http/src/routes/tasks.rs

```diff
@@ -170,6 +175,60 @@ pub struct TasksFilterQuery {
     from: Option<TaskId>,
 }
 
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct TaskDeletionQuery {
+    #[serde(rename = "type")]
+    type_: Option<CS<Kind>>,
+    uid: Option<CS<u32>>,
+    status: Option<CS<Status>>,
+    index_uid: Option<CS<IndexUid>>,
+}
+
+async fn delete_tasks(
+    index_scheduler: GuardedData<ActionPolicy<{ actions::TASKS_DELETE }>, Data<IndexScheduler>>,
+    params: web::Query<TaskDeletionQuery>,
+    _req: HttpRequest,
+    _analytics: web::Data<dyn Analytics>,
+) -> Result<HttpResponse, ResponseError> {
+    let TaskDeletionQuery {
+        type_,
+        uid,
+        status,
+        index_uid,
+    } = params.into_inner();
+
+    let kind: Option<Vec<_>> = type_.map(|x| x.into_iter().collect());
+    let uid: Option<Vec<_>> = uid.map(|x| x.into_iter().collect());
+    let status: Option<Vec<_>> = status.map(|x| x.into_iter().collect());
+    let index_uid: Option<Vec<_>> =
+        index_uid.map(|x| x.into_iter().map(|x| x.to_string()).collect());
+
+    let query = Query {
+        limit: None,
+        from: None,
+        status,
+        kind,
+        index_uid,
+        uid,
+    };
+
+    let filtered_query = filter_out_inaccessible_indexes_from_query(&index_scheduler, &query);
+
+    let tasks = index_scheduler.get_task_ids(&filtered_query)?;
+    let filtered_query_string = yaup::to_string(&filtered_query).unwrap();
+    let task_deletion = KindWithContent::TaskDeletion {
+        query: filtered_query_string,
+        tasks,
+    };
+    // TODO: Lo: analytics
```

Comment:
```suggestion
```

Currently, no analytics have been defined by gmourier I think.
We'll see later when he gives us directions on what we want to send.